slo/promql.go

```diff
@@ -552,6 +554,8 @@ func (r objectiveReplacer) replace(node parser.Node) {
 	case *parser.VectorSelector:
 		if n.Name == "errorMetric" {
 			n.Name = r.errorMetric
+		} else if n.Name == "successMetric" {
+			n.Name = r.successMetric
```

Comment:
So far it worked quite well to actually reuse the errorMetric for the other way around case of a successMetric too. It's probably possible here too. I get that naming the metric and matchers `error` isn't ideal in this case.

Reply:
Alright, I will revert this. I got a bit confused while doing this, but it's mostly because I'm not really familiar with Go overall.

Reply:
The naming is definitely not clean. The `error` should be named something else. It's really just the compare metric we compare against the total. 🤷‍♂️ Maybe that might be a good internal name at some point: `compare`.